src/main/kotlin/com/bardsoftware/backend/fleet/rmarkdown/Subscriber.kt

```diff
@@ -116,18 +115,22 @@ class TaskReceiver(tasksDirectory: String,
         if (!rootFile.exists()) {
             throw IOException("In task(id = $taskId): path to root file doesn't exists")
         }
-      
+
         return rootFile
     }
 
+    private fun compileProject(rootFileFullPath: String, zippedProject: ByteString): File {
+        // TODO: Unzip project and compile via docker
+
+        return MOCK_PDF_FILE
+    }
+
     override fun processMessage(message: PubsubMessage): Boolean {
         val request = CompilerFleet.CompilerFleetRequest.parseFrom(message.data)
         val taskId = request.taskId
         val rootFileName = request.rootFileName
-        val zipBytes = request.zipBytes
-        val rootFile = unzipCompileTask(taskId, rootFileName, zipBytes)
-        val compiledPdf = dockerProcessor.compileRmdToPdf(rootFile)
-        this.callback("Compiled pdf name: ", compiledPdf.name)
+        val zippedProject = request.zipBytes
```

Comment:
Almost certainly you will need other information from the request, such as engine name (pdflatex, xelatex, lualatex), etc., so I suggest passing the whole request to compile function. I recommend introducing a request field which indicates what compiler shall be used -- mock, real TeX, markdown => TeX converter or anything else -- and take decision here based on that field.

Reply:
OK, I'll add to request field about compiler
Do I need to pass explicitly other supporting information?(engine, etc) Or will we obtain it somehow from zipped project?

Reply:
It is passed explicitly. You can look at `backend/tex/src/main/proto/texbe.proto` file in Papeeria repository to see our current compile request and response. I believe they will eventually migrate to compiler fleet protos

Reply:
What is `NONE` in enum of `Engine` in `texbe.proto`?
I mean, should I put it in `fleet.proto` as well?

Reply:
NONE is a hack which makes our current compilers just fetching project contents but not really compiling it. For new compiling pipeline it will not make any sense